src/pudl/io_managers.py

```diff
@@ -409,6 +407,37 @@ def __init__(
         md = self.package.to_sql()
         super().__init__(base_dir, db_name, md, timeout)
 
+    def _handle_pandas_output(self, context: OutputContext, df: pd.DataFrame):
+        """Enforce PUDL DB schema and write dataframe to SQLite."""
+        table_name = self._get_table_name(context)
+        # If table_name doesn't show up in the self.md object, this will raise an error
+        sa_table = self._get_sqlalchemy_table(table_name)
+        try:
+            res = self.package.get_resource(table_name)
+        except ValueError:
+            raise ValueError(
+                f"{table_name} does not appear in pudl.metadata.resources. "
+                "Check for typos, or add the table to the metadata and recreate the "
+                f"PUDL SQlite database. It's also possible that {table_name} is one of "
+                "the tables that does not get loaded into the PUDL SQLite DB because "
+                "it's a work in progress or is distributed in Apache Parquet format."
+            )
```

Comment:
This check duplicates the `RuntimeError` raised in `sa_table = self._get_sqlalchemy_table(table_name)`. When we write dataframes to the database we need to make sure they exist in `self.md`. When we load a data frame from the database we want to make sure the table exists in has corresponding metadata in `self.package`.

Reply:
Okay, so you're saying that we'll never see this error, because the thing it's checking has already been checked by attempting to get `sa_table`?

`self.package` is the more expansive definition of the data, right? Every table that appears in `self.md` will definitely appear in `self.package`, but the reverse isn't true -- there can be Resources (like `hourly_emissions_epacems` that appear in `self.package` but not `self.md` So `_get_sqlalchemy_table()` is the more restrictive test?

I feel like we might need to add a Venn diagram to the documentation explaining the difference between:

* Resources that appear in the Package (includes DB tables and non-SQL resources like EPA CEMS... as well as views?)
* Tables that appear in the SQLAlchemy metadata generated by the Package (only includes DB tables we write to directly)
* Tables that appear in the database itself (All tables we write to directly, plus all database views)

Do I understand that correctly?

Reply:
Correct! Everything that appears in `self.md` will appear in `self.package` but not everything in `self.package` will appear in `self.md` (views, and things we don't want to the db like cems). I agree, we should clarify the difference between the metadata in `self.md` and `self.package` in the `PudlSQLiteIOManager` doc string.